app/views/subscriptions/index.html.erb

```diff
@@ -1,61 +1,67 @@
 <!--Descriptive page name, messages and instructions-->
 <h2 class="heading">
-  <%= params[:type].blank? ? ts('My Subscriptions') : 
-  ts('My %{type} Subscriptions', type: params[:type].to_s.singularize.titleize)
-  %>
+  <% if params[:type].blank? %>
+    <%= t(".page_heading.general") %>
+  <% else %>
+    <%= t(".page_heading.type", type: params[:type].to_s.singularize.titleize) %>
```

Comment:
Oh, ouch, I can't believe we've been doing it like this. It's just taking the text from the parameters and using it on the page heading, so `https://archiveofourown.org/users/sarken/subscriptions?type=fakey%20fake` will create a page that says "My Fakey Fake Subscriptions." That's not great, but it really doesn't work once we factor in translations, because even if the parameters are valid, you're going to end up with nonsense that mixes English and the desired language, e.g., "Mes Work abonnements."

That's obviously a sign of a bigger issue -- we should be checking the parameters against a list in the controller, so entering something other than `?type=series`, `?type=works`, or `type=users` will just take you to the plain subscriptions page -- that's probably out of scope.

However, since we're adding i18n, we should at least fix this heading. That means we need to stop using the `type` in the string and use it in the key instead. However, before we do that, it also means making sure the `type` is one of our valid options. Maybe something like this:

```
<% type_leaf = if Subscription::VALID_SUBSCRIBABLES.include?(params[:type].to_s.singularize.titleize)
                params[:type].to_s
              else
                "all"
              end %>
<%= t(".page_heading.#{type_leaf}") %>
```

And then the locale file would be like
```
page_heading:
  all: My Subscriptions
  works: My Work Subscriptions
  etc
```

If it's neater/less work to start the fix at the controller level, though, that's fine!

Reply:
I've tried to fix it in the controller and added some tests.